devito/types/grid.py

```diff
@@ -505,7 +527,88 @@ def define(self, dimensions):
         return {d: ('middle', 1, 1) for d in dimensions}
 
 
-class SubDomainSet(SubDomain):
+class MultiSubDimension(SubDimension):
+
+    """
+    A special SubDimension for graceful lowering of MultiSubDomains.
+    """
+
+    def __init_finalize__(self, name, parent):
+        lst, rst = self._symbolic_thickness(name)
+        left = parent.symbolic_min + lst
+        right = parent.symbolic_max - rst
```

Comment:
isn't there a -1 for the right part?

Reply:
I don't think so? this should behave like SubDimension.middle

Reply:
ok